tutorials/3.Interactions.md

```diff
@@ -0,0 +1,348 @@
+```
+
+Here we've (possibly) created an application command called "myping", with a description of "my unique ping" that is created from `myPingAction`.
+
+If we add `myPingInteraction` to the `applicationCommands` list in `pingPlugin'`, and run our bot, we can see that we can now (after a short delay) type `/myping` into discord, press enter, and get a special response in return.
+
+Well, that's pretty exciting. But we could do that before. What do we need to do to get the rest of our ping functions up to scratch?
+
+Turns out there's a fair amount of information that we took for granted before that we're now going to have to work through, but don't worry, we're gonna get through it step by step.
+
+### Labelled arguments
+
+Slash commands unfortunately need to have each of their parameters named, which means that anything we give to our function has to be named.
+
+Let's uncomment `myEcho` and roll up our sleeves to see what we want to do here.
+
+First we pull out `echoHelp` into its own top level function, modify the return and arguments so it returns a message that is just the text we pass in - and we run into an issue. Having a `RestOfInput` doesn't make any sense in a slash command, so we can't use `echoHelp` as is. We're going to have to construct two subtly different functions that we can process differently to get what we want.
+
+```haskell
+-- current `echoHelp`
+echoHelp :: RestOfInput Text -> DatabaseDiscord MessageDetails
+echoHelp (ROI t) = return $ messageDetailsBasic t
+```
+
+We take away the restriction of `RestOfInput` for now, breaking `myecho`, but we'll fix it shortly. 
+
+We then change `myEcho` so that instead of having `(parseComm echoHelp)`, we instead have `(parseComm (echoHelp . unROI))`. This function is now the same as it originally was, and we can reuse our new `echoHelp` for our interactions.
+
+```haskell
+myEcho :: Command
+myEcho = Command "myecho" (parseComm (echoHelp . unROI)) []
+
+echoHelp :: Text -> DatabaseDiscord MessageDetails
+echoHelp t = return $ messageDetailsBasic t
+```
+
+Ok, great, we're back to where we started. Now we have to make this labelled interactiony thingummy.
+
+Let's see if we can just do the same thing we just did, using the tools we had before.
+
+```haskell
+myEchoInteraction :: Maybe ApplicationCommandRecv
+myEchoInteraction = makeApplicationCommandPair "myecho" "echo your input" (echoHelp . unLabel)
+```
+
+Alright, and compi-
+
+```
+No instance for (GHC.TypeLits.KnownSymbol name0)
+        arising from a use of ‘makeApplicationCommandPair’
+    • In the expression:
+        makeApplicationCommandPair
+          "myecho" "echo your input" (echoHelp . unLabel)
+      In an equation for ‘myEchoInteraction’:
+          myEchoInteraction
+            = makeApplicationCommandPair
+                "myecho" "echo your input" (echoHelp . unLabel)
+```
+
+Oh dear that's hideous. Oh yeah, we forget to actually label anything!
```

Comment:
You've neglected to say what labelling _is_ at this point - another screenshot showing a slash command at work may help?